docs/Advanced-Usage/Heterogeneous-SoCs.rst

```diff
@@ -0,0 +1,126 @@
+Heterogeneous SoCs
+===============================
+
+The Chipyard framework involves multiple cores and accelerators that can be composed in arbitrary ways.
+This discussion will focus on how you combine Rocket, BOOM and Hwacha in particular ways to create a unique SoC.
+
+Creating a Rocket and BOOM System
+-------------------------------------------
+
+Instantiating an SoC with Rocket and BOOM cores is all done with the configuration system and two specific mixins.
+Both BOOM and Rocket have mixins labelled ``WithNBoomCores(X)`` and ``WithNBigCores(X)`` that automatically create ``X`` copies of the core.
+When used together you can create a heterogeneous system.
+The following example shows a dual core BOOM with a single core Rocket.
+
+.. code-block:: scala
+
+    class DualBoomAndOneRocketConfig extends Config(
+      new WithNormalBoomRocketTop ++
+      new WithBootROM ++
+      new boom.system.WithRenumberHarts ++
+      new boom.common.WithRVC ++
+      new boom.common.DefaultBoomConfig ++
+      new boom.system.WithNBoomCores(2) ++
+      new freechips.rocketchip.subsystem.WithoutTLMonitors ++
+      new freechips.rocketchip.subsystem.WithNBigCores(1) ++
+      new freechips.rocketchip.system.BaseConfig)
+
+In this example, the ``WithNBoomCores`` and ``WithNBigCores`` mixins set up the default parameters for the multiple BOOM and Rocket cores, respectively.
+However, for BOOM, an extra mixin called ``DefaultBoomConfig`` is added to override the default parameters with a different set of more common default parameters.
+This mixin applies to all BOOM cores in the system and changes the parameters for each.
+
+Great! Now you have a heterogeneous setup with BOOMs and Rockets.
+The final thing you need to make this system work is to renumber the ``hartId``'s of the cores so that each core has a unique ``hartId`` (a ``hartId`` is the hardware thread id of the core).
+This is done with ``WithRenumberHarts`` (which can label the Rocket cores first or the BOOM cores first).
+The reason this is needed is because by default the ``WithN...Cores(X)`` mixin assumes that there are only BOOM or only Rocket cores in the system.
+Thus, without the ``WithRenumberHarts`` mixin, each set of cores is labeled starting from zero causing multiple cores to be assigned the same ``hartId``.
+
+Another option to create a multi heterogeneous core system is to override the parameters yourself so you can specify the core parameters per core.
```

Comment:
Maybe we should emphasize that this is an alternative, and not an addition to the previous method? (I think it's clear, but it probably can't hurt to add additional separation?)